packages/x-components/src/components/base-slider.vue

```diff
@@ -0,0 +1,174 @@
+    props: {
+      threshold: {
+        type: Object as PropType<{ min: number; max: number }>,
+        default: () => ({ min: 0, max: Number.MAX_SAFE_INTEGER })
+      },
+      modelValue: {
+        type: Object as PropType<{ min: number; max: number }>,
+        required: true
+      }
+    },
+    emits: ['update:modelValue'],
+    setup(props, { emit }) {
+      let sliderInstance: API;
+      const slider = ref<HTMLElement>();
+
+      const minSelected = ref(props.modelValue?.min ?? props.threshold.min);
+      const maxSelected = ref(props.modelValue?.max ?? props.threshold.max);
+
+      const rangeSelected = computed(() => [minSelected.value, maxSelected.value]);
+      const slideRange = computed(() => ({ min: props.threshold.min, max: props.threshold.max }));
+
+      onMounted(() => {
+        sliderInstance = create(slider.value!, {
+          start: rangeSelected.value,
+          range: slideRange.value,
+          step: 1,
+          connect: true,
+          margin: 1
+        });
+
+        sliderInstance.on('update', ([min, max]) => {
+          minSelected.value = Number(min);
+          maxSelected.value = Number(max);
+        });
+
+        sliderInstance.on('change', () =>
+          emit('update:modelValue', { min: minSelected.value, max: maxSelected.value })
+        );
+      });
+
+      onUnmounted(() => {
+        // Waiting to finish the collapse animation before destroying it
+        setTimeout(sliderInstance.destroy.bind(sliderInstance), 600);
+      });
+
+      watch(
+        () => props.threshold,
+        ({ min, max }) => {
+          sliderInstance.updateOptions({ range: slideRange.value, start: [min, max] }, false);
+          emit('update:modelValue', { min, max });
+        }
+      );
+
+      watch([() => props.modelValue.min, () => props.modelValue.max], ([min, max]) => {
+        if (min === minSelected.value && max === maxSelected.value) {
+          return;
+        }
+
+        const minValidated = min < props.threshold.min ? props.threshold.min : min;
+        const maxValidated = max > props.threshold.max ? props.threshold.max : max;
+        sliderInstance.set([minValidated, maxValidated]);
+
+        if (minValidated !== min || maxValidated !== max) {
+          emit('update:modelValue', { min: minValidated, max: maxValidated });
+        }
+      });
+
+      return {
+        slider,
+        rangeSelected
+      };
+    }
+  });
+</script>
+
+<style lang="css">
+  @import 'nouislider/dist/nouislider.css';
+
+  .x-base-slider {
+    gap: 15px;
+  }
+
+  .x-base-slider,
+  .x-base-slider-selected-min,
+  .x-base-slider-selected-max {
+    display: flex;
+    justify-content: stretch;
+    align-items: stretch;
+    flex-flow: column nowrap;
+  }
+
+  .x-base-slider-selected {
+    display: inline-flex;
+    gap: 5px;
+  }
+
+  .x-base-slider-selected-min,
+  .x-base-slider-selected-max {
+    flex: 50%;
+    gap: 8px;
+  }
+
+  .x-slider {
+    height: 4px;
+    padding: 0 15px;
```

Comment:
```suggestion
    padding: 0 16px;
```

Reply:
I removed this customization, which, would need to be applied in the client app.